File: src/main/java/org/example/scalingstream/channels/DataChannelContext.java
package org.example.scalingstream.channels;

import java.util.Map;

public abstract class DataChannelContext {
  String name;
  Map<ChannelArg, Object> channelArgs;

  DataChannelContext(String name) { this.name = name; }

  DataChannelContext(String name, Map<ChannelArg, Object> channelArgs) {
    this(name);
    this.channelArgs = channelArgs;
  }

  abstract public void destroy();
}

File: src/main/java/org/example/scalingstream/channels/DataChannelContext.kt
package org.example.scalingstream.channels

abstract class DataChannelContext<Type> internal constructor(var name: String?) {
    var channelArgs: Map<ChannelArg, Any>? = null

    internal constructor(name: String?, channelArgs: Map<ChannelArg, Any>?) : this(name) {
        this.channelArgs = channelArgs
    }

    abstract fun destroy()

}